### DISCUSSION TITLE
persistent data with mongodb

### DISCUSSION THREAD

**Discussion by viniciusverasdossantos** [microsoft/aspire#4361] [Q&A] [answered]
How can I use persistent data with mongodb? every time I restart I lose all data. I didn't find where to activate data volume

**Comment by davidfowl** [comment, marked as answer]
Each database has a `WithDataVolume` method that can be used to persist data. 

https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.mongodbbuilderextensions.withdatavolume?view=dotnet-aspire-8.0&viewFallbackFrom=dotnet-aspire-8.0.1

**Reply by davidfowl** [reply]
PS: The docs have been updated https://learn.microsoft.com/en-us/dotnet/aspire/database/mongodb-integration?tabs=dotnet-cli
